### PR TITLE
Add Logic for Geomechanical Use Cases

### PR DESCRIPTION
**Pull request by bska**

### CODE REVIEW

**Comment by bska** [comment]
jenkins build this please

File: opm/elasticity/asmhandler.hpp [hunk 1]
```diff
@@ -110,7 +110,7 @@ class ASMHandler {
                     const Dune::FieldVector<double,esize>* S,
                     const LeafIterator& cell,
                     Vector* b=NULL);
-
+    void addMatElement(int i,int j,double val){ A[i][j] += val;}
```
Comment (akva2) [inline opm/elasticity/asmhandler.hpp:113]: add spaces

File: opm/elasticity/elasticity_impl.hpp [hunk 2]
```diff
@@ -71,6 +71,21 @@ void Elasticity<GridType>::getBmatrix(Dune::FieldMatrix<ctype,components,funcdim
...
+    template<class GridType>
+    template<int funcdims>
+void Elasticity<GridType>::getBVector(Dune::FieldVector<ctype,funcdims>& Bvector,
+                                      const Dune::FieldVector<ctype,dim>& point)
+{
+    P1ShapeFunctionSet<ctype,ctype,dim> basis 
+        = P1ShapeFunctionSet<ctype,ctype,dim>::instance();
+
+    Dune::FieldMatrix<ctype,funcdims,dim> N;
+    for (int i=0;i < funcdims; i++){
```
Comment (akva2) [inline opm/elasticity/elasticity_impl.hpp:84]: add space

File: opm/elasticity/meshcolorizer.hpp [hunk 3]
```diff
@@ -44,7 +44,14 @@ class MeshColorizer {
     }
 
     //! \brief Calculate the coloring
-    void calcGroups();
+    void calcGroups(){
```
Comment (akva2) [inline opm/elasticity/meshcolorizer.hpp:47]: add space

File: opm/elasticity/meshcolorizer.hpp [hunk 4]
```diff
@@ -44,7 +44,14 @@ class MeshColorizer {
...
+        tg[0].resize(1);
+        tg[0].reserve(grid.size(0));
+        for (int i = 0; i < grid.size(0); ++i)
```
Comment (akva2) [inline opm/elasticity/meshcolorizer.hpp:50]: add braces

File: opm/elasticity/meshcolorizer.hpp [hunk 5]
```diff
@@ -44,7 +44,14 @@ class MeshColorizer {
...
+            tg[0][0].push_back(i);
+        tg[1].clear();
+
```
Comment (akva2) [inline opm/elasticity/meshcolorizer.hpp:53]: remove blank line

File: opm/elasticity/meshcolorizer.hpp [hunk 6]
```diff
@@ -44,7 +44,14 @@ class MeshColorizer {
...
-    void calcGroups();
+    void calcGroups(){
+        tg[0].resize(1);
+        tg[0].reserve(grid.size(0));
+        for (int i = 0; i < grid.size(0); ++i)
+            tg[0][0].push_back(i);
+        tg[1].clear();
+      
+    }
   private:
```
Comment (akva2) [inline opm/elasticity/meshcolorizer.hpp:55]: add blank line before private

**Review by akva2 — APPROVED** [review]
Some cosmetics

**Comment by bska** [comment]
> Some cosmetics

Thank you for the review.  I've pushed an update that I believe addresses those comments, but please let me know if there are additional steps needed here.

**Comment by bska** [comment]
jenkins build this please

**Comment by akva2** [comment]
all good now afaict, so feel free to merge when green.

**Comment by bska** [comment]
> feel free to merge when green.

Thanks a lot.  The build check is green now so I'll merge into master.

**Pull request merged**